src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -132,7 +115,9 @@ public DeviceEntity call() throws Exception {
                 Root<DeviceEntity> r = query.from(DeviceEntity.class);
                 query.select(r).where(
                         criteriaBuilder.and(criteriaBuilder.equal(r.get("name"), entity.getName())));
+
```

Comment:
can we just use findDeviceByName() method here?

Reply:
@arvindn05, No I can't use the findDeviceByName, as this method retrieves from DB the device information based on Name and deviceId/MgrId. whereas in createDevice method, Id is not yet generated and so ID value is not used as it is null.

Reply:
OK, implemented.